classes/object_manipulator/manipulator.php

```diff
@@ -49,15 +49,18 @@ abstract class manipulator {
 
     protected $logger;
 
+    protected $batchsize;
+
     /**
      * Manipulator constructor
      *
      * @param object_file_system $filesystem object file system
      * @param int $maxruntime What time the file manipulator should finish execution by
      */
     public function __construct($filesystem, $config) {
-         $this->finishtime = time() + $config->maxtaskruntime;
-         $this->filesystem = $filesystem;
+        $this->finishtime = time() + $config->maxtaskruntime;
+        $this->filesystem = $filesystem;
+        $this->batchsize = $config->batchsize;
```

Comment:
Also protect this and default to 10000 if the config is not set. This is not guaranteed when you run an upgrade from the cli, it's only later that this gets set in the gui

Reply:
Also is there any current place where an admin can easily see a history of how long each cron tick took, including the time for the sql vs the time spent doing the transfer and how many it did?

This seems like a good thing to push into a moodle log record just a basic summary of this data

Reply:
Default value is set and protected on `lib.php`.

Regarding logging we can create an Issue and add it to TODO project. 
I'd suggest to store this data on a database layer. It will give us a chance to easily select the data for a period, build a graph, take a look at data dynamic, etc.